runtime/src/iree/hal/drivers/amdgpu/util/libhsa.c

```diff
@@ -0,0 +1,692 @@
+    case HSA_STATUS_ERROR_OUT_OF_RESOURCES:
+      return "HSA_STATUS_ERROR_OUT_OF_RESOURCES";
+    case HSA_STATUS_ERROR_INVALID_PACKET_FORMAT:
+      return "HSA_STATUS_ERROR_INVALID_PACKET_FORMAT";
+    case HSA_STATUS_ERROR_RESOURCE_FREE:
+      return "HSA_STATUS_ERROR_RESOURCE_FREE";
+    case HSA_STATUS_ERROR_NOT_INITIALIZED:
+      return "HSA_STATUS_ERROR_NOT_INITIALIZED";
+    case HSA_STATUS_ERROR_REFCOUNT_OVERFLOW:
+      return "HSA_STATUS_ERROR_REFCOUNT_OVERFLOW";
+    case HSA_STATUS_ERROR_INCOMPATIBLE_ARGUMENTS:
+      return "HSA_STATUS_ERROR_INCOMPATIBLE_ARGUMENTS";
+    case HSA_STATUS_ERROR_INVALID_INDEX:
+      return "HSA_STATUS_ERROR_INVALID_INDEX";
+    case HSA_STATUS_ERROR_INVALID_ISA:
+      return "HSA_STATUS_ERROR_INVALID_ISA";
+    case HSA_STATUS_ERROR_INVALID_ISA_NAME:
+      return "HSA_STATUS_ERROR_INVALID_ISA_NAME";
+    case HSA_STATUS_ERROR_INVALID_CODE_OBJECT:
+      return "HSA_STATUS_ERROR_INVALID_CODE_OBJECT";
+    case HSA_STATUS_ERROR_INVALID_EXECUTABLE:
+      return "HSA_STATUS_ERROR_INVALID_EXECUTABLE";
+    case HSA_STATUS_ERROR_FROZEN_EXECUTABLE:
+      return "HSA_STATUS_ERROR_FROZEN_EXECUTABLE";
+    case HSA_STATUS_ERROR_INVALID_SYMBOL_NAME:
+      return "HSA_STATUS_ERROR_INVALID_SYMBOL_NAME";
+    case HSA_STATUS_ERROR_VARIABLE_ALREADY_DEFINED:
+      return "HSA_STATUS_ERROR_VARIABLE_ALREADY_DEFINED";
+    case HSA_STATUS_ERROR_VARIABLE_UNDEFINED:
+      return "HSA_STATUS_ERROR_VARIABLE_UNDEFINED";
+    case HSA_STATUS_ERROR_EXCEPTION:
+      return "HSA_STATUS_ERROR_EXCEPTION";
+    case HSA_STATUS_ERROR_INVALID_CODE_SYMBOL:
+      return "HSA_STATUS_ERROR_INVALID_CODE_SYMBOL";
+    case HSA_STATUS_ERROR_INVALID_EXECUTABLE_SYMBOL:
+      return "HSA_STATUS_ERROR_INVALID_EXECUTABLE_SYMBOL";
+    case HSA_STATUS_ERROR_INVALID_FILE:
+      return "HSA_STATUS_ERROR_INVALID_FILE";
+    case HSA_STATUS_ERROR_INVALID_CODE_OBJECT_READER:
+      return "HSA_STATUS_ERROR_INVALID_CODE_OBJECT_READER";
+    case HSA_STATUS_ERROR_INVALID_CACHE:
+      return "HSA_STATUS_ERROR_INVALID_CACHE";
+    case HSA_STATUS_ERROR_INVALID_WAVEFRONT:
+      return "HSA_STATUS_ERROR_INVALID_WAVEFRONT";
+    case HSA_STATUS_ERROR_INVALID_SIGNAL_GROUP:
+      return "HSA_STATUS_ERROR_INVALID_SIGNAL_GROUP";
+    case HSA_STATUS_ERROR_INVALID_RUNTIME_STATE:
+      return "HSA_STATUS_ERROR_INVALID_RUNTIME_STATE";
+    case HSA_STATUS_ERROR_FATAL:
+      return "HSA_STATUS_ERROR_FATAL";
+  }
+}
+
+// Returns a string literal describing the given HSA status code.
+// We inline the string mapping here instead of calling hsa_status_string so
+// that we can customize the error messages to our uses (and avoid the need for
+// carrying the libhsa pointer to resolve statuses).
+static const char* iree_hsa_status_description(hsa_status_t status) {
+  switch (status) {
+    default:
+      return "Unknown.";
+    case HSA_STATUS_SUCCESS:
+      return "The function has been executed successfully.";
```

Comment:
nit: do we want to include or exclude the `.` at the end of each status description here? IIRC in compiler code we leave it off, so messages can merge together more cleanly?